library/L2_HAL/sensors/battery/test/ltc4150_test.cpp

```diff
@@ -46,28 +65,37 @@ TEST_CASE("Test LTC4150 Coulomb Counter/ Battery Gas Gauge", "[ltc4150]")
     Verify(Method(mock_backup_pol_pin, SetDirection)
                .Using(Gpio::Direction::kInput));
     Verify(Method(mock_primary_input_pin, AttachInterrupt)).Exactly(1);
+    Verify(Method(mock_primary_pol_pin, AttachInterrupt)).Exactly(1);
     Verify(Method(mock_backup_input_pin, AttachInterrupt)).Exactly(1);
+    Verify(Method(mock_backup_pol_pin, AttachInterrupt)).Exactly(1);
   }
 
   SECTION("Count Ticks and Calculate mAh")
   {
     Ltc4150 primary_counter = Ltc4150(
-        primary_isr, mock_primary_input_pin.get(), mock_primary_pol_pin.get());
-    Ltc4150 backup_counter = Ltc4150(
-        backup_isr, mock_backup_input_pin.get(), mock_backup_pol_pin.get());
-    primary_isr.IsrHandler();
-    primary_isr.IsrHandler();
-    primary_isr.IsrHandler();
-    primary_isr.IsrHandler();
-    APPROX_EQUALITY(primary_counter.GetBatterymAh(), 0.68271f, kResolution);
+        mock_primary_input_pin.get(), mock_primary_pol_pin.get(), resistance);
+    Ltc4150 backup_counter = Ltc4150(mock_backup_input_pin.get(),
+                                     mock_backup_pol_pin.get(),
+                                     resistance,
+                                     Ltc4150::Polarity::kCharging);
 
-    backup_isr.IsrHandler();
-    backup_isr.IsrHandler();
-    backup_isr.IsrHandler();
-    backup_isr.IsrHandler();
-    backup_isr.IsrHandler();
-    backup_isr.IsrHandler();
-    APPROX_EQUALITY(backup_counter.GetBatterymAh(), 1.02407f, kResolution);
+    primary_counter.Initialize();
+    primary_input_isr();
+    primary_input_isr();
+    primary_input_isr();
+    primary_input_isr();
+    primary_pol_isr();
+    CHECK(primary_counter.GetCharge().to<float>() == Approx(-0.068271f));
```

Comment:
Can you derive the approximate values you use here so that they do not look like mere magic numbers. Same with the stuff below.

Reply:
Done.